static_frame/core/frame.py

```diff
@@ -1393,6 +1393,51 @@ def from_json(cls,
                 consolidate_blocks=consolidate_blocks
                 )
 
+    @classmethod
+    @doc_inject(selector='constructor_frame')
+    def from_msgpack(cls,
+            msgpack_data: bin,
+            ) -> 'Frame':
+        '''Frame constructor from an in-memory binary object formatted as a msgpack.
+
+        Args:
+            msgpack_data: a binary msgpack object, encoding a static_frame as produced from Frame.to_msgpack.
+
+        Returns:
+            :obj:`static_frame.Frame`
+        '''
+        import msgpack
+        import msgpack_numpy
+        def uncast_msgpack(input):
+            data = msgpack.unpackb(input, object_hook=msgpack_numpy.decode) #try coercing numpy datatypes
+            if not data or isinstance(data, dict):
+                return data
+                
+            #This is the parser portion of the magic character hack to handle any arbitrary datatypes
+            #TODO: Is this attempting to handle too much? Is there a better way?
+            if isinstance(data[0], str) and data[0][0] == '>': #if we detect the magic character, evaluate class
+                import sys
+                evaldata = []
+                clsname = data[0][1:]
+                m, c = clsname.split('.',1)
+                cls = getattr(sys.modules[m], c)
+                #cls = eval(clsname) #replacement code above, leaving this in so that it's obvious what's going on.
+                
+                for d in data[1:]:
+                    evaldata.append(cls.__new__(cls, d))
+                return evaldata
+            return data
+        unpacked = uncast_msgpack(msgpack_data)
+        for key in ['_name','_index','_columns']:
+            unpacked[key] = uncast_msgpack(unpacked[key])
+        for i in range(len(unpacked['_blocks'])):
+            unpacked['_blocks'][i] = uncast_msgpack(unpacked['_blocks'][i])
+        
+        return cls.from_records(unpacked['_blocks'],
+                columns=unpacked['_columns'],
+                index=unpacked['_index'],
+                name=unpacked['_name'])
```

Comment:
I think you are using `from_records` here because you transposed the `Frame`, below. Better to not transpose, and create a `TypeBlocks` instance directly from `TypeBlocks.from_blocks`, then feed that to `Frame.__init__` constructor setting `own_data=True`.

Reply:
The other way around actually, I used transpose to make this work. Still bad :P. I found TypeBlocks, but didn't figure from_blocks yet.

Reply:
from_msgpack now returns instance using TypeBlocks.from_blocks setting own_data=True